src/qt/optionsdialog.cpp

```diff
@@ -164,10 +196,122 @@ void OptionsDialog::disableApply()
+{
+    // Map model to widgets
+    mapper->addMapping(fee_edit, OptionsModel::Fee);
+    mapper->addMapping(detach_database, OptionsModel::DetachDatabases);
+}
+
+/* Display options */
+DisplayOptionsPage::DisplayOptionsPage(QWidget *parent):
+    OptionsPage(parent), restart_warning_displayed(false)
+{
+    setWindowTitle(tr("Display"));
+
+    QVBoxLayout *layout = new QVBoxLayout();
+
+    QHBoxLayout *lang_hbox = new QHBoxLayout();
+    lang_hbox->addSpacing(18);
+    QLabel *lang_label = new QLabel(tr("User Interface &Language: "));
+    lang_hbox->addWidget(lang_label);
+    lang = new QValueComboBox(this);
+    // Make list of languages
+    QDir translations(":translations");
+    lang->addItem("(default)", QVariant(""));
+    foreach(const QString &langStr, translations.entryList())
+    {
+       lang->addItem(langStr, QVariant(langStr));
+    }
+
+    lang->setToolTip(tr("The user interface language can be set here. This setting will only take effect after restarting Paycoin."));
+    connect(lang, SIGNAL(activated(int)), this, SLOT(showRestartWarning()));
+
+    lang_label->setBuddy(lang);
+    lang_hbox->addWidget(lang);
+
+    layout->addLayout(lang_hbox);
+
+    QHBoxLayout *unit_hbox = new QHBoxLayout();
+    unit_hbox->addSpacing(18);
+    QLabel *unit_label = new QLabel(tr("&Unit to show amounts in: "));
+    unit_hbox->addWidget(unit_label);
+    unit = new QValueComboBox(this);
+    unit->setModel(new BitcoinUnits(this));
+    unit->setToolTip(tr("Choose the default subdivision unit to show in the interface, and when sending coins"));
+
+    unit_label->setBuddy(unit);
+    unit_hbox->addWidget(unit);
+
+    layout->addLayout(unit_hbox);
+
+    display_addresses = new QCheckBox(tr("&Display addresses in transaction list"), this);
+    display_addresses->setToolTip(tr("Whether to show Paycoin addresses in the transaction list"));
+    layout->addWidget(display_addresses);
+
+    coin_control_features = new QCheckBox(tr("Display coin control features (experts only!)"), this);
+    coin_control_features->setToolTip(tr("Whether to show coin control features or not"));
+    layout->addWidget(coin_control_features);
+
+     layout->addStretch();
+     setLayout(layout);
+ }
+
+ void DisplayOptionsPage::setMapper(MonitoredDataMapper *mapper)
+ {
+     mapper->addMapping(lang, OptionsModel::Language);
+     mapper->addMapping(unit, OptionsModel::DisplayUnit);
+     mapper->addMapping(display_addresses, OptionsModel::DisplayAddresses);
+     mapper->addMapping(coin_control_features, OptionsModel::CoinControlFeatures);
+}
+
+void DisplayOptionsPage::showRestartWarning()
+{
+    if(!restart_warning_displayed)
+    {
+        QMessageBox::warning(this, tr("Warning"), tr("This setting will take effect after restarting Paycoin."), QMessageBox::Ok);
+        restart_warning_displayed = true;
+    }
+>>>>>>> 45232d8... Make it possible to set user interface language from options dialog
```

Comment:
Whoops. If I said yes I'd be lying :P I'll fix this now and properly test myself!